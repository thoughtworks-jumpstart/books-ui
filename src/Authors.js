import React from "react";

const Authors = ({ data = [] }) => {
  return (
    <div>
      <h1>Authors</h1>
      {data.map(author => {
        return <li key={author._id}>{author.name}</li>;
      })}
    </div>
  );
};

export default Authors;
