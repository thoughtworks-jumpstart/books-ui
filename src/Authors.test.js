import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Authors from "./Authors.js";

it("renders a list of authors", () => {
  const authors = [
    {
      name: "J K Rowling",
      _id: "123"
    },
    {
      name: "George R R Martin",
      _id: "456"
    }
  ];

  const renderer = new ShallowRenderer();
  renderer.render(<Authors data={authors} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
