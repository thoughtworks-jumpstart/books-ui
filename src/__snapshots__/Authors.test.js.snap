// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of authors 1`] = `
<div>
  <h1>
    Authors
  </h1>
  <li>
    J K Rowling
  </li>
  <li>
    George R R Martin
  </li>
</div>
`;
